contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,690 @@
+
+        UtilsWrapperInterface.ActionArgs[] memory actions = new UtilsWrapperInterface.ActionArgs[](3);
+
+        actions[0] = UtilsWrapperInterface.ActionArgs(
+            UtilsWrapperInterface.ActionType.OpenVault,
+            address(this), // owner
+            address(this), // receiver
+            address(0), // asset, otoken
+            vaultID, // vaultId
+            0, // amount
+            0, // index
+            abi.encode(1) // vault type
+        );
+
+        // Approve margin pool to deposit collateral
+        IERC20(_collateralAsset).safeApproveNonCompliant(addressbook.getMarginPool(), totalInitialMargin);
+
+        // deposit collateral
+        actions[1] = UtilsWrapperInterface.ActionArgs(
+            UtilsWrapperInterface.ActionType.DepositCollateral,
+            address(this), // owner
+            address(this), // address to transfer from
+            _collateralAsset, // deposited asset
+            vaultID, // vaultId
+            totalInitialMargin, // amount
+            0, //index
+            "" //data
+        );
+
+        // create otoken, get an address and calculate mint amount
+        address otoken = IOtokenFactoryWrapperInterface(addressbook.getOtokenFactory()).createOtoken(
+            order.underlying,
+            USDC,
+            _collateralAsset,
+            order.strikePrice,
+            order.expiry,
+            order.isPut
+        );
+
+        // scales by 1e8 for division with oracle price
+        // scales by 1e2 to increase from 6 decimals (notional) to 8 decimals (otoken)
+        uint256 mintAmount = order.notional.mul(1e10).div(oracle.getPrice(order.underlying));
+
+        actions[2] = UtilsWrapperInterface.ActionArgs(
+            UtilsWrapperInterface.ActionType.MintShortOption,
+            address(this), // owner
+            address(this), // address to transfer to
+            otoken, // option address
+            vaultID, // vaultId
+            mintAmount, // amount
+            0, // index
+            "" // data
+        );
+
+        // execute actions
+        controller.operate(actions);
+
+        // transfer otokens to user
+        IERC20(otoken).safeTransfer(order.buyer, mintAmount);
+
+        // order accounting
+        orders[_orderID].premium = _premium;
+        orders[_orderID].collateral = _collateralAsset;
+        orders[_orderID].seller = msg.sender;
+        orders[_orderID].vaultID = vaultID;
+        orders[_orderID].oToken = otoken;
+        orderStatus[_orderID] = OrderStatus.Succeeded;
+
+        emit OrderExecuted(_orderID, _collateralAsset, _premium, msg.sender, vaultID, otoken, totalInitialMargin);
+    }
+
+    /**
+     * @notice settles the vault after expiry
+     * @dev can only be called by the market maker who is the order seller
+     * @param _orderID order id
+     */
+    function settleVault(uint256 _orderID) external nonReentrant {
+        require(_orderID <= latestOrder, "OTCWrapper: inexistent order");
```

Comment:
same as [this](https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2/files#r1136446876) comment

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2/commits/f6866eef4da54cca9b8c6ee51c0e10b435284aa4